Mathlib/Algebra/FreeMonoid/Basic.lean

```diff
@@ -407,4 +407,31 @@ theorem length_reverse {a : FreeMonoid α} : a.reverse.length = a.length :=
 
 end Reverse
 
+section IsomorphicTypes
+
+variable {α β : Type*} (e : α ≃ β)
+
+/-- free monoids over isomorphic types are isomorphic -/
+@[to_additive "if two types are isomorphic, the additive free monoids over those types are
+isomorphic"]
+def congr_iso : FreeMonoid α ≃* FreeMonoid β :=
+  MulEquiv.mk' ⟨FreeMonoid.map e.toFun, FreeMonoid.map e.invFun, fun x => by simp [map_map],
+    fun x => by simp [map_map]⟩ (by simp [map_mul])
```

Comment:
Maybe you can add this theorems and use them in the proof:
```lean
theorem map_infFun_map_toFun_eq {x : FreeMonoid α} : (map e.invFun) (map e.toFun x) = x := sorry
theorem map_toFun_map_invFun_eq {x : FreeMonoid β} : (map e.toFun) (map e.invFun x) = x := sorry
```

Reply:
good idea, thanks!